### PR TITLE
Topic recognition - S44889160

### PR DESCRIPTION
**Pull request by josh-lim1234**
Hi Shakes,

Here is my pull request for the COMP3710 report.

The submission contains:
* graphs
* .gitignore
* lateral_classification.py
* main.py
* README.md

### graphs
The graphs directory has two images in it. 'accuracy.png' was created by lateral_classification.py and 'lastepochstat.png' is a screenshot.
### .gitignore
The gitignore file is to exclude the directory 'img' from being pushed which contains the OAI AKOA knee dataset. Note that if there is no 'img' dataset within the 'AKOA_Analysis' directory containing all the images from https://cloudstor.aarnet.edu.au/sender/?s=download&token=d82346d9-f3ca-48bf-825f-327a622bfaca (link from blackboard) the model will not have any training data.
### lateral_classification.py
This file contains the code that creates and trains the model,  code that generates the 'accuracy.png' file seen in the README.md file and splits the training data to create the validation set. 
### main.py
This file contains the code that imports and creates labels for the data based on the name of the files in the 'img' directory. The file also splits the data into a training and test set. The 'train' function from 'lateral_classification.py' is called from here.
### README.md
Contains explanation of solution.

Thank you,
Josh 
SID: 44889160

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Commit messages should be more specific
- Code valid for Easy Difficulty
- Results are good

Other Feedback
- Would like to see some example MR classification visualised in README

**Comment by rexsw** [comment]
Code looks good for easy difficulty

For next time;

lateral _classification.py doesn’t need a main function if it isn’t going to be run

Tensorflow has functions for image resizing and more so there’s no need to also use PIL in your project.

File: recognition/AKOA_Analysis/lateral_classification.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import matplotlib as ml
```
Comment (shakes76) [inline recognition/AKOA_Analysis/lateral_classification.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme minimal
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Resolve conflict on the Recognition folder readme file.**
